### PR TITLE
Change update transmitter balance behaviour

### PR DESCRIPTION
**Pull request by infiloop2**

### CODE REVIEW

File: contracts/src/v0.8/dev/automation/2_1/KeeperRegistryBase2_1.sol [hunk 1]
```diff
@@ -659,14 +659,13 @@ abstract contract KeeperRegistryBase2_1 is ConfirmedOwner, ExecutionPrevention {
...
-    transmitter.lastCollected = totalPremium;
-
-    // Transfer spare change to owner
-    s_storage.ownerLinkBalance += (uncollected - due * payeeCount);
-    s_transmitters[transmitterAddress] = transmitter;
+    if (transmitter.active) {
+      uint96 uncollected = totalPremium - transmitter.lastCollected;
+      uint96 due = uncollected / payeeCount;
+      transmitter.balance += due;
+      transmitter.lastCollected = totalPremium; // transmitter forfeits some spare change
```
Comment (RyanRHall) [inline contracts/src/v0.8/dev/automation/2_1/KeeperRegistryBase2_1.sol:666]: could we do...

```solidity
transmitter.lastCollected = totalPremium - (totalPremium%payeeCount);
```

I think that will allow the "spare change" to roll over to the next withdraw.

File: contracts/test/v0.8/automation/KeeperRegistry2_1.test.ts [hunk 2]
```diff
@@ -4978,4 +4981,217 @@ describe('KeeperRegistry2_1', () => {
...
+      const k2New = await registry.getTransmitterInfo(
+        await keeper2.getAddress(),
+      )
+
+      // transmitter info lastCollected should be updated for k1, not for k2
+      assert.isTrue(k1New.lastCollected.eq(registryPremium))
+      assert.isTrue(k2New.lastCollected.eq(BigNumber.from(0)))
+    })
+
+    const verifyConsistentAccounting = async (
```
Comment (RyanRHall) [inline contracts/test/v0.8/automation/KeeperRegistry2_1.test.ts:5072]: nit - I would move this to the top of the file with all the other helper functions

Comment (RyanRHall) [reply contracts/test/v0.8/automation/KeeperRegistry2_1.test.ts:5072]: this is a great test tho

File: contracts/src/v0.8/dev/automation/2_1/KeeperRegistryBase2_1.sol (hunk 1 above)
Comment (infiloop2) [reply contracts/src/v0.8/dev/automation/2_1/KeeperRegistryBase2_1.sol]: done

**Review by FelixFan1992 — APPROVED** [review]

**Review by FelixFan1992 — APPROVED** [review]

**Pull request merged**